src/cubie/integrators/loops/ode_loop.py

```diff
@@ -440,7 +440,7 @@ def loop_fn(
                         do_save = (t + dt[0]  +equality_breaker) >= next_save
                         dt_eff = selp(do_save, next_save - t, dt[0])
```

Comment:
there is an inconsistency here, as dt[0] is in precision type. this will likely be upcast in the sum, then the dt_eff will be selecting between a float64 and precision object, which will fail in CUDA mode. Cast next_save - t to precision in the dt_eff selp, as dt_eff should be in "precision" units to match step size.


```suggestion
do_save = (t + dt[0]  +equality_breaker) >= next_save
dt_eff = selp(do_save, precision(next_save - t), dt[0])
```